File: Day 065/2483. Minimum Penalty for a Shop.java
class Solution {
    public int bestClosingTime(String customers) {
        int s = 0;
        int ms = 0;     
        int ans = -1;   

        for (int i = 0; i < customers.length(); i++) {
            if (customers.charAt(i) == 'Y') {
                s += 1;
            } else {
                s -= 1;
            }
            if (s > ms) {
                ms = s;
                ans = i;
            }
        }
        return ans + 1;
    }
}

File: Day 065/2483. Minimum Penalty for a Shop.cpp
class Solution {
public:
    int bestClosingTime(string customers) {
        int s =0;
        int ms =0;
        int ans =-1;
        for (int i =0;i<customers.size();i++){
            if (customers[i]== 'Y'){
                s+= 1;
            }
            else{
                s-= 1;
            }
            if (s>ms){
                ms = s;
                ans = i;
            }
        }
        return ans+1;
    }
};